examples/ui/toolbar.rs

```diff
@@ -0,0 +1,279 @@
+                    ButtonTags::Stop => None,
+                    ButtonTags::Play => Some("Playing...".to_string()),
+                    ButtonTags::Pause => Some("Pause...".to_string()),
+                };
+
+                ev_toolbar_button.send(ToolbarEvent {
+                    tag: button_tag.clone(),
+                    message,
+                    interaction: interaction.clone(),
+                });
+            }
+            Interaction::Hovered => {
+                ev_toolbar_button.send(ToolbarEvent {
+                    tag: button_tag.clone(),
+                    message: None,
+                    interaction: interaction.clone(),
+                });
+            }
+            Interaction::None => {
+                ev_toolbar_button.send(ToolbarEvent {
+                    tag: button_tag.clone(),
+                    message: None,
+                    interaction: interaction.clone(),
+                });
+            }
+        }
+    }
+}
+
+fn toolbar_state_system(
+    mut ev_toolbar_button: EventReader<ToolbarEvent>,
+    mut toolbar_state: ResMut<State<ToolbarState>>,
+) {
+    for ev in ev_toolbar_button.iter() {
+        if ev.interaction == Interaction::Clicked {
+            if ev.tag != toolbar_state.current().selected {
+                println!("Toolbar state: Setting {:?} as selected", ev.tag);
+                toolbar_state
+                    .set(ToolbarState {
+                        selected: ev.tag.clone(),
+                    })
+                    .unwrap();
+            }
+        }
+    }
+}
+
+fn toolbar_style_system(
+    mut ev_toolbar_button: EventReader<ToolbarEvent>,
+    mut style_query: Query<(&mut Style, &mut UiColor, &ButtonTags), With<Button>>,
+    toolbar_state: Res<State<ToolbarState>>,
+) {
+    for ev in ev_toolbar_button.iter() {
+        for (mut style, mut color, tag) in &mut style_query {
+            // Update button colours
+            let this_button = ev.tag.clone() == tag.clone();
+            let hovered = this_button && ev.interaction == Interaction::Hovered;
+            let clicked = this_button && ev.interaction == Interaction::Clicked;
+            let selected = tag.clone() == toolbar_state.current().selected;
+
+            // println!(
+            //     "Checking {:?} hovered?: {:?} clicked: {:?} selected: {:?}",
+            //     tag, hovered, clicked, selected
+            // );
+
+            if selected {
+                if clicked {
+                    *color = PRESSED_BUTTON.into();
+                } else {
+                    *color = SELECTED_BUTTON.into();
+                }
+            } else {
+                if clicked {
+                    *color = PRESSED_BUTTON.into();
+                } else if hovered {
+                    *color = HOVERED_BUTTON.into();
+                } else {
+                    *color = NORMAL_BUTTON.into();
+                }
+            }
+
+            if tag.clone() == ButtonTags::Stop
```

Comment:
Why clone? Couldn't you just do:
```suggestion
            if tag == &ButtonTags::Stop
```